### ISSUE TITLE
Force refresh?

### ISSUE THREAD

**Issue by chriscn** [franc6/ics_calendar#82]
Is there a way to force refresh of the calendar subscriptions? Currently I just restart Home Assistant

Registering a service would be great. :-)

**Comment by franc6** [comment]
That's something controlled almost entirely by HA.  There are two items in this platform which affect refreshing data.  First, is a restriction of calling an update method more than once every 15 minutes.  That's hard-coded, and matches what the calendar platforms that come with HA do.  I've noticed that HA doesn't call the function nearly that often, and that frequency, is, AFAIK, hard-coded within HA (I haven't looked in a couple of years, so that may have changed).  I have no control over when the update function is called, and calling it outside of the proper context (i.e. the HA code that calls it, and also the only thing a service that's part of this platform could do) isn't useful, because whatever you're looking at in HA won't be updated.
The second thing is an option to control the download frequency, which exists mainly to prevent downloading too often, thereby reducing resource usage.  See [README.md](../blob/releases/README.md#download-interval) for more info.
Unfortunately, changing either of those things won't provide any kind of on-demand refresh.

So, basically, ask in one of the HA forums, maybe someone will add a service in HA to do this.  I don't think you're the only one.  I know that when I want to test this platform in HA, it's a pain to have to wait for HA to decide to update things. :(
If such a thing is added, I'll probably have to make some changes to be compatible with it.
